Bindings/MainWindow.xaml

```diff
@@ -16,27 +32,46 @@
             <ColumnDefinition Width="*"/>
         </Grid.ColumnDefinitions>
         <Grid.RowDefinitions>
+            <RowDefinition MaxHeight="30"/>
             <RowDefinition MaxHeight="30"/>
             <RowDefinition/>
         </Grid.RowDefinitions>
+        <CheckBox
+            Grid.Column="0"
+            Grid.Row="0"
+            Grid.ColumnSpan="6"
+            Content="Display math fields"
+            IsThreeState="False"
+            IsChecked="{Binding MathEnabled}"
+            />
         <TextBox 
             Grid.Column="0"
+            Grid.Row="1"
+            Visibility="{Binding MathEnabled, Converter={StaticResource VisibilityConverter}}"
```

Comment:
An alternative here wold be to keep the old grid and set visibility on it. What you have works but is a slight n-plication.